mava/wrappers/observation.py

```diff
@@ -81,18 +81,17 @@ def observation_spec(
         """Specification of the observation of the `RobotWarehouse` environment."""
         obs_spec = self._env.observation_spec()
         num_obs_features = obs_spec.agents_view.shape[-1] + self._env.num_agents
+        dtype = obs_spec.agents_view.dtype
 
-        agents_view = specs.Array(
-            (self._env.num_agents, num_obs_features), jnp.int32, "agents_view"
-        )
+        agents_view = specs.Array((self._env.num_agents, num_obs_features), dtype, "agents_view")
 
         if self.has_global_state:
             wrapped_state_shape = obs_spec.global_state.shape
             state_shape = (
                 *wrapped_state_shape[:-1],
                 wrapped_state_shape[-1] + self._env.num_agents,
             )
-            global_state = specs.Array(state_shape, jnp.int32, "global_state")
+            global_state = specs.Array(state_shape, dtype, "global_state")
```

Comment:
Quite a nit question, but can't there be weird cases where a global state has a different dtype than the agent's view?